MCPClient: Add dashboard & archivematicaCommon to sys.path

Hwesta opened the pull request:
MCPClient adds the dashboard and archivematicaCommon to the sys.path for all client scripts.  Remove all changes to sys.path from client scripts.

Feedback on this both as a feature and implementation approach welcome.

Hwesta committed:
MCPClient: Read DJANGO_SETTINGS_MODULE from config

Read the DJANGO_SETTINGS_MODULE from a config file in MCPClient. This
will also set DJANGO_SETTINGS_MODULE for all client scripts.

Remove unused default MCPArchivematicaServerInterface.

mistydemeo commented on src/MCPClient/lib/clientScripts/elasticSearchIndexProcessAIP.py:
```diff
@@ -21,9 +21,9 @@
-sys.path.append("/usr/lib/archivematica/archivematicaCommon/externals")
+
+# archivematicaCommon
+# externals
```
I actually think `externals` is no longer required here.

mistydemeo replied:
Ignore that, I forgot about `xmltodict`. Probably don't need to comment it here though?

mistydemeo commented:
:+1:

mistydemeo commented:
Instead of mutating the parent thread's own environment, maybe it's worth pushing this into the `executeOrRun` call in archivematicaClient.py? We could provide a generic way to specify a dict that will be merged into the subprocess's environment, and make the call of `executeOrRun` pass an environment dict with the `PYTHONPATH` and `DJANGO_SETTINGS_MODULE`.

Hwesta commented:
A-ha! I hadn't spotted the executeOrRun call, and thought everything had to be done in the threading calls and startThread.  I like that idea better.

DJANGO_SETTINGS_MODULE will still have to be set in archivematicaClient.py, because it imports from the dashboard.

Hwesta committed:
MCPClient: Add dashboard & archivematicaCommon to sys.path

MCPClient adds the dashboard and archivematicaCommon to the sys.path
for all client scripts.

mistydemeo commented:
Looks good! :+1:  I'm vaguely iffy on `env_updates` as the kwarg name, but I don't have a concrete suggestion. Going to merge this as part of #124?

Hwesta commented:
No, it just came out of that work but was conceptually separate.  Make the boilerplate for setting up Django in each client script shorter if I don't have to set the path and settings module every time.
